### PR TITLE
chore: add code coverage

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add code coverage reporting.

Relates to https://github.com/arcus-azure/arcus/issues/264

### CODE REVIEW

File: src/Arcus.EventGrid.Testing/Logging/NoOpLogger.cs [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
 
 namespace Arcus.EventGrid.Testing.Logging
 {
     /// <summary>
     /// Represents an <see cref="ILogger"/> model without any implementation.
     /// </summary>
+    [Obsolete("Unused logger implementation with the same functionality as " + nameof(NullLogger) + " will be removed in future major")]
     public class NoOpLogger : ILogger
```
Comment (fgheysels) [inline src/Arcus.EventGrid.Testing/Logging/NoOpLogger.cs:11]: It looks like the code-coverage reports leads us to types that we can mark as obsolete :)

File: src/Arcus.EventGrid.Tests.Integration/Publishing/EventGridPublisherClientWithTrackingEventGridEventTests.cs [hunk 2]
```diff
@@ -124,47 +124,5 @@ private static void AssertEventGridEventForData(EventGridEvent expected, EventGr
             string actual = endpoint.ConsumerHost.GetReceivedEventOrFail(expected.Id);
             ArcusAssert.ReceivedNewCarRegisteredEvent(expected, actual);
         }
-
```
Comment (fgheysels) [inline src/Arcus.EventGrid.Tests.Integration/Publishing/EventGridPublisherClientWithTrackingEventGridEventTests.cs:127]: Tests that are being removed after applying coverage reporting; interesting :)

**Review by fgheysels — APPROVED** [review]

File: src/Arcus.EventGrid.Tests.Integration/Publishing/EventGridPublisherClientWithTrackingEventGridEventTests.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.EventGrid.Tests.Integration/Publishing/EventGridPublisherClientWithTrackingEventGridEventTests.cs:127]: These tests were actually unit tests that were hiding in the integration test project. Moved them.

File: src/Arcus.EventGrid.Testing/Logging/NoOpLogger.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.EventGrid.Testing/Logging/NoOpLogger.cs:11]: Yes, these two loggers in this project does not have much added-value and it seems too big a work to maintain them.

**Pull request merged**